### ISSUE TITLE
gccgo: inconsistent handling of float constants rounded to 0.0

### ISSUE THREAD

**Issue by griesemer** [golang/go#12621]
The following program

```
package main
import "fmt"
var p = +1e-10000
var m = -1e-10000
func main() {
    fmt.Println(p < 0, m < 0)
    fmt.Println(p == 0, m == 0)
    fmt.Println(p > 0, m > 0)
    fmt.Println(p, m)
}
```

prints

```
false false
true true
false false
0 -0
```

when executing after compiling with gccgo (gccgo (google-gccgo-219769) 5.0.0 20150116 (experimental)) . That is, while "ideal" constant arithmetic pretends that both p and m are perfectly 0, when the values are assigned to a variable, m becomes -0. This seems odd.

I would expect that floating-point constants that are 0 in the ideal constant expression world at compile time never convert to a value with a negative sign.

See also #12576.

**Comment by paranoiacblack** [comment]
pinging myself to revisit this since #12576 has been resolved

**Comment by ianlancetaylor** [comment]
Patch had to be rolled back because it used MPFR functions not available in 2.4.2.  Reopening.
